examples/hpc/dla-driver/src/lib.rs

```diff
@@ -483,46 +530,62 @@ impl Dla {
         }
     }
 
+    /// Reads index of the first input bank
     fn get_input_bank(&self) -> MemoryBank {
         let mut reg = self.read_u32(DLA_BUF_DATA_BANK);
         reg = get_bits!(reg, DLA_BUF_DATA_BANK_B_BITMASK);
-        MemoryBank::from_u32(reg)
+        let bank = match MemoryBank::try_from(reg) {
+            Ok(bank) => bank,
+            Err(e) => panic!("{}", e),
+        };
+        bank
     }
 
+    /// Reads index of the first kernel bank
     fn get_kernel_bank(&self) -> MemoryBank {
         let mut reg = self.read_u32(DLA_BUF_DATA_BANK);
         reg = get_bits!(reg, DLA_BUF_DATA_BANK_A_BITMASK);
-        MemoryBank::from_u32(reg)
+        let bank = match MemoryBank::try_from(reg) {
+            Ok(bank) => bank,
+            Err(e) => panic!("{}", e),
+        };
+        bank
     }
 
+    /// Reads index of the first output bank
     fn get_output_bank(&self) -> MemoryBank {
         let reg = self.read_u32(DLA_PP_AXI_WRITE);
         let bank_idx: u32 = (reg - MEMORY_BANK_BASE_ADDR as u32) / MEMORY_BANK_SIZE as u32;
-        MemoryBank::from_u32(bank_idx)
+        let bank = match MemoryBank::try_from(bank_idx) {
+            Ok(bank) => bank,
+            Err(e) => panic!("{}", e),
+        };
+        bank
     }
 
+    /// Sets clipping after conv2d
     fn set_mac_clip(&self, clip_amount: u32) {
-        let mut reg = self.read_u32(DLA_MAC_CTRL);
         // Cap clipping amount
         if clip_amount > 21 {
-            reg = set_bits!(DLA_MAC_CLIP_OFFSET, DLA_MAC_CLIP_BITMASK, reg, 0x1F);
-        } else {
-            reg = set_bits!(DLA_MAC_CLIP_OFFSET, DLA_MAC_CLIP_BITMASK, reg, clip_amount);
+            panic!("Clip can't be more than 20")
         }
+        let mut reg = self.read_u32(DLA_MAC_CTRL);
+        reg = set_bits!(DLA_MAC_CLIP_OFFSET, DLA_MAC_CLIP_BITMASK, reg, clip_amount);
         self.write_u32(DLA_MAC_CTRL, reg)
     }
 
+    /// Sets clipping after post-processing
     fn set_pp_clip(&self, clip_amount: u32) {
-        let mut reg = self.read_u32(DLA_PP_CTRL);
         // Cap clipping amount
         if clip_amount > 0x1F {
-            reg = set_bits!(DLA_PP_CLIP_OFFSET, DLA_PP_CLIP_BITMASK, reg, 0x1F);
-        } else {
-            reg = set_bits!(DLA_PP_CLIP_OFFSET, DLA_PP_CLIP_BITMASK, reg, clip_amount);
+            panic!("Clip can't be more than 20")
```

Comment:
Consider returning `Result<(), InvalidClip(u32)>`, as strings and panics are not always the most convenient choice for debugging at this level. This way the correct display format can be chosen at the user level.

Reply:
Both clip setting functions now return `Result<(), InvalidClip>`